### PR TITLE
add elvToken and update redeemable

### PR DESCRIPTION
**Pull request by elv-preethi**
* Added `ElvToken` contract and updated `Redeemable` contract to use new access control.
* `ElvToken` is similar to TestToken (initially it has certain amount minted) and it inherits pausable, erc20Burnable and accessControlEnumerable.
* Changed `Erc20PaymentsTest` to use `ElvToken` instead of TestToken.

### CODE REVIEW

File: src/tradable/cmn/Roles.sol [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+library ElvCmnRoles {
```
Comment (elv-serban) [inline src/tradable/cmn/Roles.sol:4]: Is there a benefit in adding the subdirectory `cmn/`?  Would there be a roles file that is not common inside `tradable/`

Comment (elv-preethi) [reply src/tradable/cmn/Roles.sol:4]: `Redeemable.sol` uses minter_role as well so added as common roles.

Do you want me to keep it separate?

File: src/payments/ERC20Payments.sol [hunk 2]
```diff
@@ -15,6 +15,7 @@ contract ERC20Payments {
         Canceled
```
Comment (elv-serban) [inline src/payments/ERC20Payments.sol:15]: Should be called "Claimed"

File: src/payments/ERC20Payments.sol [hunk 3]
```diff
@@ -94,52 +92,49 @@ contract ERC20Payments {
...
-    onlyOracle(_refId)
-    returns (bool)
+        external
+        contractExists(_refId)
+        withdrawable(_refId)
+        onlyOracle(_refId)
+        returns (bool)
     {
         LockPayment storage c = paymentTransactions[_refId];
         c.state = PaymentState.Withdrawn;
```
Comment (elv-serban) [inline src/payments/ERC20Payments.sol]: To move assignment after or just before emit Claimed(_paymentId)

File: src/tradable/Redeemable.sol [hunk 4]
```diff
@@ -15,14 +16,13 @@ import "src/MinterRole.sol";
  * have already been removed.
  * Redemption state is stored in one 256 bit bitmap per tokenId.
  */
-contract Redeemable is MinterRole {
-
+contract Redeemable is AccessControlEnumerable {
     event RedeemableAdded(uint8 offerId);
     event RedeemableRemoved(uint8 offerId);
-    event Redeem(address redeemer, uint256 tokenId, uint offerId);
```
Comment (elv-serban) [inline src/tradable/Redeemable.sol:22]: offerId must be 8 bits because we are using a 'bitmap'

**Review by elv-serban — APPROVED** [review]
Looks good!

**Pull request merged**